### PR TITLE
fix: duplicating user with same email when ldap login

### PR DESCRIPTION
**Pull request by Aleksgav**
- Fix duplicating user with same email when ldap login
- Fix wrong ldap request for username or email login
- Another minor fixes

### CODE REVIEW

File: object/ldap_conn.go [hunk 1]
```diff
@@ -376,6 +404,11 @@ func SyncLdapUsers(owner string, syncUsers []LdapUser, ldapId string) (existUser
 	}
 	tag := strings.Join(ou, ".")
 
+	existingEmails, err := GetExistingEmails(owner, emails)
```
Comment (yamaritta) [inline object/ldap_conn.go:407]: This logic should only work when turned  application.EnableLinkWithEmail

File: object/ldap_conn.go [hunk 2]
```diff
@@ -544,12 +594,14 @@ func SyncUserFromLdap(organization string, ldapId string, userName string, passw
 		return nil, err
 	}
 
-	user := &User{
-		Name: userName,
+	user := &User{}
+	if strings.Contains(userName, "@") {
```
Comment (yamaritta) [inline object/ldap_conn.go:598]: maybe better regex mail rule

File: object/ldap_conn.go [hunk 3]
```diff
@@ -559,20 +611,31 @@ func SyncUserFromLdap(organization string, ldapId string, userName string, passw
...
 		if err != nil {
+			logs.Error("check ldap user password: %s", err.Error())
+
 			conn.Close()
 			return nil, err
 		}
 
 		_, _, err = SyncLdapUsers(organization, AutoAdjustLdapUser(res), ldapServer.Id)
+		if err != nil {
+			logs.Error("sync ldap users: %s", err.Error())
```
Comment (yamaritta) [inline object/ldap_conn.go:635]: Is added error logging written to records?  it’s not very good if some of errors is in log and some is in records

File: object/ldap_conn.go (hunk 2 above)
Comment (Aleksgav) [reply object/ldap_conn.go]: done

File: object/ldap_conn.go (hunk 3 above)
Comment (Aleksgav) [reply object/ldap_conn.go:643]: logs is not equal records
logging is located in many places for debugging and understanding
Records only in places when it necessary
In this place record needed?

File: object/ldap_conn.go (hunk 1 above)
Comment (Aleksgav) [reply object/ldap_conn.go:409]: What about background sync?

**Pull request closed without merging**